prow/plank/controller.go

```diff
@@ -442,7 +447,19 @@ func (c *Controller) startPod(pj kube.ProwJob) (string, string, error) {
 				Name:  "BUILD_NUMBER",
 				Value: buildID,
 			},
+			kube.EnvVar{
+				Name:  "NAME",
+				Value: pj.Metadata.Name,
```

Comment:
IE this is the prow generated UUID name for the job / pod for a particular run, not the job's name from the spec.